src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -62,125 +58,32 @@ def frames_from_metadata(metadata_path: Path) -> dict[int, FrameMeta]:
-        sw_cumul_disp_xr.attrs = granule_xr.attrs
-    sw_cumul_disp_xr.attrs['bbox'] = bbox
-    return sw_cumul_disp_xr
-
-
-def update_reference_date(granule: xr.DataArray, frame: FrameMeta) -> xr.DataArray:
-    """Update the reference date of a granule to match the reference date of a frame.
-
-    Note: Assumes that reference date that will be updated is older than the current reference date.
-
-    Args:
-        granule: The granule to update
-        frame: The frame metadata
-    """
-    fully_updated = utils.within_one_day(granule.attrs['reference_date'], frame.reference_date)
-    while not fully_updated:
-        if granule.attrs['reference_date'] < frame.reference_date:
-            raise ValueError('Granule reference date is older than frame reference date, cannot be updated.')
-        prev_ref_date = granule.attrs['reference_date']
-        prev_ref_date_min = prev_ref_date - timedelta(days=1)
-        prev_ref_date_max = prev_ref_date + timedelta(days=1)
-        # We can assume that there is only one granule for a frame that has
-        # a secondary date equal to another granule's reference date
-        granule_dict = find_needed_granules(
-            [frame.frame_id], prev_ref_date_min, prev_ref_date_max, strategy='max', min_granules=1
-        )
-        older_granule_meta = granule_dict[frame.frame_id][0]
-        older_granule = load_sw_disp_granule(older_granule_meta, granule.attrs['bbox'])
-        granule += older_granule
-        granule.attrs['reference_date'] = older_granule.attrs['reference_date']
-        fully_updated = utils.within_one_day(granule.attrs['reference_date'], frame.reference_date)
-
-    return granule
-
-
 def add_granule_data_to_array(
-    granule: Granule, frame: FrameMeta, frame_map: np.ndarray, geotransform: Affine, sw_cumul_disp: np.ndarray
+    frame_id: FrameMeta,
+    frame_map: np.ndarray,
+    geotransform: Affine,
+    begin_date: datetime,
+    end_date: datetime,
+    sw_cumul_disp: np.ndarray,
 ) -> tuple[np.ndarray, str]:
     """Add granule data to the short wavelength cumulative displacement array
 
     Args:
-        granule: The granule to add
-        frame: The frame metadata
+        frame_id: The frame id
         frame_map: The frame map array
         geotransform: The geotransform of the frame map (Rasterio style)
+        begin_date: The start of the date range
+        end_date: The end of the date range
         sw_cumul_disp: The short wavelength cumulative displacement array
 
     Returns:
         The updated short wavelength cumulative displacement array and the secondary date of the granule
     """
     bbox = utils.create_buffered_bbox(geotransform.to_gdal(), frame_map.shape, 120)  # EPSG:3857 is in meters
-    sw_cumul_disp_xr = load_sw_disp_granule(granule, bbox)
-    update_reference_date(sw_cumul_disp_xr, frame)
-
+    sw_cumul_disp_xr = load_sw_disp_stack(frame_id, bbox, begin_date, end_date, 'spanning')[-1]
```

Comment:
There's potential to improve memory usage here, but I'm content not to worry about that until we have a concrete need to optimize. Currently we're loading the whole spanning set into memory at once, then discarding all but the last. Keeping only a running total in memory would give equivalent results.